pkg/services/publicdashboards/service/query.go

```diff
@@ -280,11 +281,19 @@ func extractQueriesFromPanels(panels []interface{}, result map[int64][]*simplejs
 		}
 
 		var panelQueries []*simplejson.Json
+		hasExpression := panelHasAnExpression(panel)
```

Comment:
Just a question, to understand the underlying flow of expression. Is it possible for expressions to evaluate hidden queries?

Reply:
Yes, an expression like `$A + $B` could be displayed even when queries `A` and `B` were hidden. This is why we can't remove them in advance before calculating the expression.